include/re_types.h

```diff
@@ -263,9 +263,13 @@ typedef bool _Bool;
 #define likely(x)       __builtin_expect(!!(x), 1)
 #define unlikely(x)     __builtin_expect(!!(x), 0)
 
-
+/* Socket helpers */
 #ifdef WIN32
 #define ERRNO_SOCKET WSAGetLastError()
+#define BAD_SOCKET INVALID_SOCKET
```

Comment:
perhaps add a nice prefix to "BAD_SOCKET" ?

UDP_BAD_SOCKET
NET_BAD_SOCKET
SOCK_BAD_SOCKET
etc.

Reply:
Will do some research to complete this list:

https://github.com/baresip/re/wiki/Socket-Handling

So we have a complete overview which return types/values are used.

Reply:
It looks like only `socket` and `accept` use INVALID_SOCKET (`BAD_SOCK`) on win32. So I think we can keep the (< 0) error checks for the rest and replace only `errno` with `ERRNO_SOCK`